src/main/kotlin/org/rust/ide/intentions/AddDerive.kt

```diff
@@ -31,12 +31,12 @@ class AddDerive : PsiElementBaseIntentionAction()
             return false
         }
         val item = element.parentOfType<RustItem>() ?: return false
-        return getStructOrEnumKeyword(item) != null
+        return getLeftmostStructOrEnumKeyword(item) != null
     }
 
-    private fun getStructOrEnumKeyword(item: RustItem): PsiElement? = when (item) {
-        is RustStructItem -> item.struct
-        is RustEnumItem   -> item.enum
+    private fun getLeftmostStructOrEnumKeyword(item: RustItem): PsiElement? =when (item) {
```

Comment:
Thanks, missed it.

As for `firstChild`, I'm not sure it would work. It would mean inserting the new attribute before all other attributes, for example. I'd personally prefer it to be the "inmost" one.